src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -306,3 +315,227 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
+                                     config['sasTestHarnessConfigs'][index]['hostName'],
+                                     config['sasTestHarnessConfigs'][index]['port'],
+                                     config['sasTestHarnessConfigs'][index]['serverCert'],
+                                     config['sasTestHarnessConfigs'][index]['serverKey'],
+                                     config['sasTestHarnessConfigs'][index]['caCert']))
+
+    for index, test_harness in enumerate(sas_test_harness_objects):
+        # Start the server
+        test_harness.start()
+
+        # informing SAS UUT about SAS Test Harnesses
+        certificate_hash = getCertificateFingerprint(config['sasTestHarnessConfigs'][index]['serverCert'])
+        self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                       'url': test_harness.getBaseUrl()})
+
+    # Step 4,5 : Inject IAP protected entities into UUT
+    for iteration_content in config['iterationData']:
+      records = iteration_content['protectedEntities']
+      if records.has_key('fssRecords'):
+        for fss_record in records['fssRecords']:
+          self._sas_admin.InjectFss({'record': fss_record})
+      if records.has_key('gwpzRecords'):
+        for gwpz_record in records['gwpzRecords']:
+          self._sas_admin.InjectWisp({'record': gwpz_record})
+      if records.has_key('escRecords'):
+        for esc_record in records['escRecords']:
+          self._sas_admin.InjectEscSensorDataRecord({'record': esc_record})
+      if records.has_key('ppaRecords'):
+        for ppa_record in records['ppaRecords']:
+          self._sas_admin.InjectZoneData({'record': ppa_record})
+
+      # Step 6,7 : Creating FAD Object and Pull FAD records from SAS UUT
+      if len(iteration_content['sasTestHarnessData']) > 0:
+        fad_uut_object = getFullActivityDumpSasUut(self._sas,
+                                                   self._sas_admin,
+                                                   config['domainProxyConfigs'][0]['cert'],
+                                                   config['domainProxyConfigs'][0]['key'])
+
+        # Pull FAD from SAS Test Harnesses
+        for index, test_harness in enumerate(sas_test_harness_objects):
+          fad_test_harnesses_objects.append(getFullActivityDumpSasTestHarness(
+                                  test_harness.getSasTestHarnessInterface(),
+                                  config['sasTestHarnessConfigs'][index]['serverCert'],
+                                  config['sasTestHarnessConfigs'][index]['serverKey']
+                              )
+          )
+
+      # Step 8 : Trigger CPAS and wait until completion
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+      # Step 9 : Call IAP reference model, calculated at step 12
+      # Collect all the protected Entites
+      protected_entities.append(iteration_content['protectedEntities'])
```

Comment:
Is this really the format you want? I don't think it's the format used in e.g. the pre IAP ref model.

Reply:
This portion of code is used for pre-IAP functionality. Removed now. We will add new code once pre-IAP model is integrated.